includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1312,7 +1312,7 @@ private function is_valid_layout( $tag_spec, $node ) {
 		$allow_fluid  = Layout::FLUID === $layout_attr;
 		$allow_auto   = true;
 
-		$input_width = new CssLength( $node->getAttribute( Attribute::WIDTH ) );
+		$input_width = new CssLength( $node->hasAttribute( Attribute::WIDTH ) ? $node->getAttribute( Attribute::WIDTH ) : null );
```

Comment:
Good, as `getAttribute()` for a missing attribute returns an empty string in PHP.

This will better correspond with how the AMP JS Validator does:

```
  const widthAttr = attrsByKey['width'];
  // ...
  const inputWidth = new CssLength(
      widthAttr, /* allowAuto */ true,
      /* allowFluid */ inputLayout === generated.AmpLayout.Layout.FLUID);
```

Where in JS the `widthAttr` will be `undefined` if the attribute is absent.

https://github.com/ampproject/amphtml/blob/406f2883379377c98a14e645c8f42a9f7794bfa7/validator/js/engine/validator.js#L4533-L4535